_posts/2016-10-24-deploy-jekyll-with-travis.md

```diff
@@ -0,0 +1,232 @@
+
+That version of `travis.yml` worked for us when we were using `jekyll build` to
+compile the site, but things got trickier when we started using gulp. We had to
+set the proper versions of npm and node.js, and ensure that our ruby gems were
+installed properly. Here's all the travis config we needed to get things
+running with gulp:
+
+```yaml
+# travis.yml
+
+language: node_js
+sudo: required
+script:
+  - ./_scripts/run-tests.sh
+branches:
+  except:
+  - master
+node_js:
+ - '6.1'
+before_install:
+  - if [[ `npm -v` != 3* ]]; then npm i -g npm@3; fi
+  - rvm install 2.1
+  - rvm use 2.1
+  - . $HOME/.nvm/nvm.sh && nvm install 6.1 && nvm use 6.1
+  - gem install bundler
+  - bundle check || bundle install
+env:
+  global:
+  - NOKOGIRI_USE_SYSTEM_LIBRARIES=true
+  - secure: [stuff]
+  - secure: [more stuff]
+deploy:
+  provider: script
+  script: "./_scripts/build.sh"
+  skip_cleanup: false
+  on:
+    branch: source
+```
+
+### The deployment script
+
+When the `source` branch is updated and our tests pass, the deployment script is
+executed.
+
+```sh
+#!/bin/bash
+
+# Enable error reporting to the console.
+set -e
+
+# Install bundles if needed.
+bundle check || bundle install
+
+# NPM install if needed.
+. $HOME/.nvm/nvm.sh && nvm install 6.1 && nvm use 6.1
+npm install
+
+# Build the site.
+gulp
+
+# Checkout `master` and remove everything
+git clone https://${GH_TOKEN}@github.com/savaslabs/savaslabs.github.io.git ../savaslabs.github.io.master
+cd ../savaslabs.github.io.master
+git checkout master
+rm -rf *
+
+# Copy generated HTML site from source branch in original repo.
+# Now the `master` branch will contain only the contents of the _site directory.
+cp -R ../savaslabs.github.io/_site/* .
+
+# Make sure we have the updated .travis.yml file so tests won't run on master.
+cp ../savaslabs.github.io/.travis.yml .
+git config user.email ${GH_EMAIL}
+git config user.name "savas-bot"
+
+# Commit and push generated content to `master` branch.
+git status
+git add -A .
+git status
+git commit -a -m "Travis #$TRAVIS_BUILD_NUMBER"
+git push --quiet origin `master` > /dev/null 2>&1
+```
+
+Some important notes:
+
+- Make sure the script file is executable!
+- The last line ensures that our encrypted token won't end up in git logs
+
+## Deployment and debugging
+
+Merging this code into `source` was the easiest way for us to test if our new process worked
+— scary, right? Fortunately, if the build script failed (and it did several
+times until we could get the npm and node versions correct and the ruby gems
+installing properly) nothing got pushed to `master` so nothing was deployed,
+meaning no downtime on our site. We also used
+[Teleconsole](https://www.teleconsole.com/) to debug from inside the Travis
+environment.
+
+To do this, we commented out the `git push` line and added the following to
+`build.sh`:
+
```

Comment:
New paragraph break at the Teleconsole section and combine the 2 following sentences together?
> ...no downtime on our site.
>
> We also used Teleconsole to debug from inside the Travis environment. To do this....